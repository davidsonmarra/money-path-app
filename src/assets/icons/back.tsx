import React from 'react';
import Svg, {Path} from 'react-native-svg';

interface Props {
  size?: number;
  color?: string;
}

const BackIcon = ({size = 24, color = '#FFF'}: Props) => (
  <Svg
    width={size}
    height={size}
    fill={color}
    viewBox="0 -960 960 960"
    testID="icon-back">
    <Path d="M360-240 120-480l240-240 56 56-144 144h568v80H272l144 144z" />
  </Svg>
);

export default BackIcon;
