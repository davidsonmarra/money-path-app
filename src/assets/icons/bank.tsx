import React from 'react';
import Svg, {Path} from 'react-native-svg';

interface Props {
  size?: number;
  color?: string;
}

const BankIcon = ({size = 24, color = '#FFF'}: Props) => (
  <Svg
    width={size}
    height={size}
    fill={color}
    viewBox="0 -960 960 960"
    testID="icon-bank">
    <Path d="M200-280v-280h80v280zm240 0v-280h80v280zM80-120v-80h800v80zm600-160v-280h80v280zM80-640v-80l400-200 400 200v80zm178-80h444zm0 0h444L480-830z" />
  </Svg>
);

export default BankIcon;
