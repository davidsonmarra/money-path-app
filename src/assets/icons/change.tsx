import React from 'react';
import Svg, { Path } from 'react-native-svg';

interface Props {
  size?: number;
  color?: string;
}

const ChangeIcon = ({ size = 24, color = '#FFF' }: Props) => (
  <Svg
    width={size}
    height={size}
    fill={color}
    viewBox="0 -960 960 960"
    testID="icon-change"
  >
    <Path d="m482-200 114-113-114-113-42 42 43 43q-28 1-54.5-9T381-381q-20-20-30.5-46T340-479q0-17 4.5-34t12.5-33l-44-44q-17 25-25 53t-8 57q0 38 15 75t44 66 65 43.5 74 15.5l-38 38zm165-170q17-25 25-53t8-57q0-38-14.5-75.5T622-622t-65.5-43-74.5-14l38-39-42-42-114 113 114 113 42-42-44-44q27 0 55 10.5t48 30.5 30.5 46 10.5 52q0 17-4.5 34T603-414zM480-80q-83 0-156-31.5T197-197t-85.5-127T80-480t31.5-156T197-763t127-85.5T480-880t156 31.5T763-763t85.5 127T880-480t-31.5 156T763-197t-127 85.5T480-80m0-80q134 0 227-93t93-227-93-227-227-93-227 93-93 227 93 227 227 93m0-320" />
  </Svg>
);

export default ChangeIcon;
