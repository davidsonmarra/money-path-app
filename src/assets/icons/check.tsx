import React from 'react';
import Svg, {Path} from 'react-native-svg';

interface Props {
  size?: number;
  color?: string;
}

const CheckIcon = ({size = 24, color = '#FFF'}: Props) => (
  <Svg
    width={size}
    height={size}
    fill={color}
    viewBox="0 -960 960 960"
    testID="icon-check">
    <Path d="m424-296 282-282-56-56-226 226-114-114-56 56zm56 216q-83 0-156-31.5T197-197t-85.5-127T80-480t31.5-156T197-763t127-85.5T480-880t156 31.5T763-763t85.5 127T880-480t-31.5 156T763-197t-127 85.5T480-80m0-80q134 0 227-93t93-227-93-227-227-93-227 93-93 227 93 227 227 93m0-320" />
  </Svg>
);

export default CheckIcon;
