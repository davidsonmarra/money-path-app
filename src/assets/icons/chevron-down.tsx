import React from 'react';
import Svg, { Path } from 'react-native-svg';

interface Props {
  size?: number;
  color?: string;
}

const ChevronDownIcon = ({ size = 24, color = '#000' }: Props) => (
  <Svg
    width={size}
    height={size}
    fill={color}
    viewBox="0 -960 960 960"
    testID="icon-chevron-down"
  >
    <Path d="M480-345 240-585l56-56 184 184 184-184 56 56-240 240z" />
  </Svg>
);

export default ChevronDownIcon;
