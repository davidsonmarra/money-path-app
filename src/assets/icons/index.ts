export {default as AppleIcon} from 'src/assets/icons/apple';
export {default as BackIcon} from 'src/assets/icons/back';
export {default as BancoDoBrasilIcon} from 'src/assets/icons/banco-do-brasil';
export {default as BankIcon} from 'src/assets/icons/bank';
export {default as CheckIcon} from 'src/assets/icons/check';
export {default as ChevronRightIcon} from 'src/assets/icons/chevron-right';
export {default as CloseIcon} from 'src/assets/icons/close';
export {default as ErrorIcon} from 'src/assets/icons/error';
export {default as GoogleIcon} from 'src/assets/icons/google';
export {default as InterIcon} from 'src/assets/icons/inter';
export {default as ItauIcon} from 'src/assets/icons/itau';
export {default as MoneyPathIcon} from 'src/assets/icons/money-path';
export {default as NubankIcon} from 'src/assets/icons/nubank';
export {default as SantanderIcon} from 'src/assets/icons/santander';
export {default as WalletIcon} from 'src/assets/icons/wallet';
export {default as WarningIcon} from 'src/assets/icons/warning';
