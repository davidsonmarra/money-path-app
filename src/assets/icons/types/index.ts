export enum IconType {
  apple = 'apple',
  back = 'back',
  bancoDoBrasil = 'bancoDoBrasil',
  bank = 'bank',
  check = 'check',
  chevronRight = 'chevronRight',
  close = 'close',
  error = 'error',
  food = 'food',
  games = 'games',
  google = 'google',
  health = 'health',
  help = 'help',
  house = 'house',
  inter = 'inter',
  itau = 'itau',
  moneyPath = 'moneyPath',
  money = 'money',
  nubank = 'nubank',
  santander = 'santander',
  transport = 'transport',
  wallet = 'wallet',
  warning = 'warning',
}
