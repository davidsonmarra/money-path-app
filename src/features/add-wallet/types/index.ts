import {IconType} from 'src/assets/icons/types';

export interface WalletProps {
  name: string;
  icon: IconType;
  color?: string;
  backgroundColor?: string;
  type: 'personal' | 'bank';
  amount?: number;
}

export type SelectWalletSection = {
  title: string;
  data: WalletProps[];
};
