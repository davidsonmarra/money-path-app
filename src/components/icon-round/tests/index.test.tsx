import React from 'react';
import { render } from '@testing-library/react-native';
import IconRound, { Props as IconRoundProps } from 'src/components/icon-round';
import getStyles from 'src/components/icon-round/styles';
import { IconType } from 'src/assets/icons/types';

type Props = Partial<IconRoundProps>;

const mockIcon = IconType.wallet;
const renderComponent = ({ icon = mockIcon, ...rest }: Props) =>
  render(<IconRound icon={icon} {...rest} />);

describe('IconRound', () => {
  it('renders correctly', () => {
    const tree = renderComponent({}).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with default props correctly', () => {
    const iconRound = renderComponent({}).root;
    expect(iconRound.props.style).toStrictEqual([
      getStyles({}).container,
      { backgroundColor: undefined },
    ]);
  });

  it('renders with custom backgroundColor correctly', () => {
    const backgroundColor = '#FF0000';
    const iconRound = renderComponent({ backgroundColor }).root;
    expect(iconRound.props.style).toStrictEqual([
      getStyles({}).container,
      { backgroundColor },
    ]);
  });

  it('renders with custom size correctly', () => {
    const size = 24;
    renderComponent({ size });
    expect(renderComponent({ size })).toBeTruthy();
  });

  it('renders with custom color correctly', () => {
    const color = '#00FF00';
    renderComponent({ color });
    expect(renderComponent({ color })).toBeTruthy();
  });

  it('renders with different icon types correctly', () => {
    const iconTypes = [IconType.bank, IconType.money, IconType.wallet];

    iconTypes.forEach(iconType => {
      const tree = renderComponent({ icon: iconType }).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  it('renders with default roundSize correctly', () => {
    const iconRound = renderComponent({}).root;
    expect(iconRound.props.style).toStrictEqual([
      getStyles({}).container,
      { backgroundColor: undefined },
    ]);
  });

  it('renders with custom roundSize correctly', () => {
    const roundSize = 64;
    const iconRound = renderComponent({ roundSize }).root;
    expect(iconRound.props.style).toStrictEqual([
      getStyles({ roundSize }).container,
      { backgroundColor: undefined },
    ]);
  });

  it('renders with small roundSize correctly', () => {
    const roundSize = 24;
    const iconRound = renderComponent({ roundSize }).root;
    expect(iconRound.props.style).toStrictEqual([
      getStyles({ roundSize }).container,
      { backgroundColor: undefined },
    ]);
  });

  it('renders with large roundSize correctly', () => {
    const roundSize = 96;
    const iconRound = renderComponent({ roundSize }).root;
    expect(iconRound.props.style).toStrictEqual([
      getStyles({ roundSize }).container,
      { backgroundColor: undefined },
    ]);
  });
});
