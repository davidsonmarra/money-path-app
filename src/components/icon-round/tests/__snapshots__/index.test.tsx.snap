// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconRound renders correctly 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "borderRadius": 24,
        "height": 48,
        "justifyContent": "center",
        "width": 48,
      },
      {
        "backgroundColor": undefined,
      },
    ]
  }
>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight={32}
    bbWidth={32}
    fill="#EAC43D"
    focusable={false}
    height={32}
    meetOrSlice={0}
    minX={0}
    minY={-960}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        {
          "flex": 0,
          "height": 32,
          "width": 32,
        },
      ]
    }
    testID="icon-wallet"
    vbHeight={960}
    vbWidth={960}
    width={32}
  >
    <RNSVGGroup
      fill={
        {
          "payload": 4293575741,
          "type": 0,
        }
      }
      propList={
        [
          "fill",
        ]
      }
    >
      <RNSVGPath
        d="M240-160q-66 0-113-47T80-320v-320q0-66 47-113t113-47h480q66 0 113 47t47 113v320q0 66-47 113t-113 47zm0-480h480q22 0 42 5t38 16v-21q0-33-23.5-56.5T720-720H240q-33 0-56.5 23.5T160-640v21q18-11 38-16t42-5m-74 130 445 108q9 2 18 0t17-8l139-116q-11-15-28-24.5t-37-9.5H240q-26 0-45.5 13.5T166-510"
        fill={
          {
            "payload": 4278190080,
            "type": 0,
          }
        }
      />
    </RNSVGGroup>
  </RNSVGSvgView>
</View>
`;

exports[`IconRound renders with different icon types correctly 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "borderRadius": 24,
        "height": 48,
        "justifyContent": "center",
        "width": 48,
      },
      {
        "backgroundColor": undefined,
      },
    ]
  }
>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight={32}
    bbWidth={32}
    fill="#EAC43D"
    focusable={false}
    height={32}
    meetOrSlice={0}
    minX={0}
    minY={-960}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        {
          "flex": 0,
          "height": 32,
          "width": 32,
        },
      ]
    }
    testID="icon-bank"
    vbHeight={960}
    vbWidth={960}
    width={32}
  >
    <RNSVGGroup
      fill={
        {
          "payload": 4293575741,
          "type": 0,
        }
      }
      propList={
        [
          "fill",
        ]
      }
    >
      <RNSVGPath
        d="M200-280v-280h80v280zm240 0v-280h80v280zM80-120v-80h800v80zm600-160v-280h80v280zM80-640v-80l400-200 400 200v80zm178-80h444zm0 0h444L480-830z"
        fill={
          {
            "payload": 4278190080,
            "type": 0,
          }
        }
      />
    </RNSVGGroup>
  </RNSVGSvgView>
</View>
`;

exports[`IconRound renders with different icon types correctly 2`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "borderRadius": 24,
        "height": 48,
        "justifyContent": "center",
        "width": 48,
      },
      {
        "backgroundColor": undefined,
      },
    ]
  }
>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight={32}
    bbWidth={32}
    fill="#EAC43D"
    focusable={false}
    height={32}
    meetOrSlice={0}
    minX={0}
    minY={-960}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        {
          "flex": 0,
          "height": 32,
          "width": 32,
        },
      ]
    }
    testID="icon-money"
    vbHeight={960}
    vbWidth={960}
    width={32}
  >
    <RNSVGGroup
      fill={
        {
          "payload": 4293575741,
          "type": 0,
        }
      }
      propList={
        [
          "fill",
        ]
      }
    >
      <RNSVGPath
        d="M441-120v-86q-53-12-91.5-46T293-348l74-30q15 48 44.5 73t77.5 25q41 0 69.5-18.5T587-356q0-35-22-55.5T463-458q-86-27-118-64.5T313-614q0-65 42-101t86-41v-84h80v84q50 8 82.5 36.5T651-650l-74 32q-12-32-34-48t-60-16q-44 0-67 19.5T393-614q0 33 30 52t104 40q69 20 104.5 63.5T667-358q0 71-42 108t-104 46v84z"
        fill={
          {
            "payload": 4278190080,
            "type": 0,
          }
        }
      />
    </RNSVGGroup>
  </RNSVGSvgView>
</View>
`;

exports[`IconRound renders with different icon types correctly 3`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "borderRadius": 24,
        "height": 48,
        "justifyContent": "center",
        "width": 48,
      },
      {
        "backgroundColor": undefined,
      },
    ]
  }
>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight={32}
    bbWidth={32}
    fill="#EAC43D"
    focusable={false}
    height={32}
    meetOrSlice={0}
    minX={0}
    minY={-960}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        {
          "flex": 0,
          "height": 32,
          "width": 32,
        },
      ]
    }
    testID="icon-wallet"
    vbHeight={960}
    vbWidth={960}
    width={32}
  >
    <RNSVGGroup
      fill={
        {
          "payload": 4293575741,
          "type": 0,
        }
      }
      propList={
        [
          "fill",
        ]
      }
    >
      <RNSVGPath
        d="M240-160q-66 0-113-47T80-320v-320q0-66 47-113t113-47h480q66 0 113 47t47 113v320q0 66-47 113t-113 47zm0-480h480q22 0 42 5t38 16v-21q0-33-23.5-56.5T720-720H240q-33 0-56.5 23.5T160-640v21q18-11 38-16t42-5m-74 130 445 108q9 2 18 0t17-8l139-116q-11-15-28-24.5t-37-9.5H240q-26 0-45.5 13.5T166-510"
        fill={
          {
            "payload": 4278190080,
            "type": 0,
          }
        }
      />
    </RNSVGGroup>
  </RNSVGSvgView>
</View>
`;
